zkevm-circuits/src/witness/rw.rs

```diff
@@ -38,6 +41,61 @@ impl RwMap {
             debug_assert_eq!(idx, rw_counter - 1);
         }
     }
+    /// Check value in the same way like StateCircuit
+    pub fn check_value(&self) {
+        let mock_rand = Fr::from(0x1000u64);
+        let err_msg_first = "first access reads don't change value";
+        let err_msg_non_first = "non-first access reads don't change value";
+        let rows = self.table_assignments();
+        let updates = MptUpdates::mock_from(&rows);
+        let mut errs = Vec::new();
+        for idx in 1..rows.len() {
+            let row = &rows[idx];
+            let prev_row = &rows[idx - 1];
+            let is_first = {
+                let key = |row: &Rw| {
+                    (
+                        row.tag() as u64,
+                        row.id().unwrap_or_default(),
+                        row.address().unwrap_or_default(),
+                        row.field_tag().unwrap_or_default(),
+                        row.storage_key().unwrap_or_default(),
+                    )
+                };
+                key(prev_row) != key(row)
+            };
+            if !row.is_write() {
+                let value = row.value_assignment::<Fr>(mock_rand);
+                if is_first {
+                    // value == init_value
+                    let init_value = updates
+                        .get(row)
+                        .map(|u| u.value_assignments(mock_rand).1)
+                        .unwrap_or_default();
+                    if value != init_value {
+                        errs.push((idx, err_msg_first, *row, *prev_row));
+                    }
+                } else {
+                    // value == prev_value
+                    let prev_value = prev_row.value_assignment::<Fr>(mock_rand);
+
+                    if value != prev_value {
+                        errs.push((idx, err_msg_non_first, *row, *prev_row));
+                    }
+                }
+            }
+        }
+        log::debug!("after rw value check, err num: {}", errs.len());
+        for (idx, err_msg, row, prev_row) in errs {
+            log::debug!(
```

Comment:
Did you consider using `log::error!` instead of `log::debug!` for this line and the previous, so that it shows up as a more important log?